src/dotnet/UI.Blazor/Components/ContextMenu/BlazorContextMenu/Components/ContextMenuTrigger.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Net.Http;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Routing;
+using BlazorContextMenu.Services;
+using Microsoft.JSInterop;
+using Microsoft.AspNetCore.Components.RenderTree;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace BlazorContextMenu
+{
+    [SuppressMessage("Usage", "MA0011:IFormatProvider is missing")]
+    [SuppressMessage("Usage", "MA0074:Avoid implicit culture-sensitive methods")]
+    public class ContextMenuTrigger : ComponentBase, IDisposable
+    {
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            /*
+                <div @attributes="Attributes"
+                     onclick="@(MouseButtonTrigger == MouseButtonTrigger.Left || MouseButtonTrigger == MouseButtonTrigger.Both ? $"blazorContextMenu.OnContextMenu(event, '{MenuId.Replace("'","\\'")}'); " : "")"
+                     ondblclick="@(MouseButtonTrigger == MouseButtonTrigger.DoubleClick ? $"blazorContextMenu.OnContextMenu(event, '{MenuId.Replace("'","\\'")}'); " : "")"
+                     oncontextmenu="@(MouseButtonTrigger == MouseButtonTrigger.Right || MouseButtonTrigger == MouseButtonTrigger.Both ? $"blazorContextMenu.OnContextMenu(event, '{MenuId.Replace("'","\\'")}');": "")"
+                     class="@CssClass">
+                    @ChildContent
+                </div>
+             */
+
+            builder.OpenElement(0, WrapperTag);
```

Comment:
Maybe we migrate this code to .razor? IMO manual codegen worth it only if you can't solve the problem w/ authomatic one.

Reply:
It adds different event handlers depending on value of MouseButtonTrigger property. So it makes sense to keep it.

Reply:
removed